### PR TITLE
refactor!(aws): Inline enum extensions

### PR DESCRIPTION
**Pull request by dnys1**
commit-id:aa7ed814

---

**Stack**:
- #1625
- #1624
- #1623
- #1621
- #1620 ⬅
- #1613

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

File: packages/aws_signature_v4/example/pubspec_overrides.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# This file is checked in until it can be easily generated locally
+# and in CI pipelines, e.g. with melos.
+#
+# It may cause issues with publishing until https://github.com/dart-lang/pub/pull/3419
+# is released in stable.
+dependency_overrides:
+  aws_common:
+    path: ../../aws_common
```
Comment (Jordan-Nelson) [inline packages/aws_signature_v4/example/pubspec_overrides.yaml:8]: Does melos not already use the local version?

Comment (dnys1) [reply packages/aws_signature_v4/example/pubspec_overrides.yaml:8]: melos does, but mono_repo runs `dart pub upgrade` as its first step, which will break melos

Comment (Jordan-Nelson) [reply packages/aws_signature_v4/example/pubspec_overrides.yaml:8]: Oh I see. It looks like melos is moving toward using this method of overriding dependencies locally and in the future will generate this?

Comment (Jordan-Nelson) [reply packages/aws_signature_v4/example/pubspec_overrides.yaml:8]: Okay. It looks like melos can generate these. Do we want to switch over to that? It appears to be a simple update to melos.yaml. I'm not sure if we would want to then gitignore all the generate files or commit them.

```yaml
command:
  bootstrap:
    usePubspecOverrides: true # pubspec_overrides.yaml will be autogenerate during bootstrap
```

Comment (dnys1) [reply packages/aws_signature_v4/example/pubspec_overrides.yaml:8]: Exactly, yep! I tried that and it doesn't work for our repo :/ Still need to investigate more.

https://github.com/aws-amplify/amplify-flutter/pull/1585

**Review by Jordan-Nelson — APPROVED** [review]

**Review by HuiSF — APPROVED** [review]

**Pull request merged**